app/scripts/models/reliers/oauth.js

```diff
@@ -140,10 +146,10 @@ define(function (require, exports, module) {
       self._importRequiredSearchParam('client_id', 'clientId');
```

Comment:
What do you think about validating `client_id`, `access_type`, and possibly `scope` as part of this work?

`state` is supposed to be opaque to us, so ignore that. If we whitelist what is allowed in `scope`, then we'll have to change that validation logic every time a new scope is possible. I'm not sure if we want to deal with that. @rfk?

`access_type` can only be `online` and `offline`.

`client_id` - if the format is a constant, we should validate it.